fluent/src/types.js

```diff
@@ -45,8 +45,12 @@ export class FluentType {
 }
 
 export class FluentNone extends FluentType {
+  valueOf() {
+    return null;
+  }
+
   toString() {
-    return this.value || "???";
+    return `{${this.value || "???"}}`;
   }
 }
```

Comment:
For what I mentioned earlier, I ended up with https://github.com/projectfluent/fluent-rs/blob/master/fluent-bundle/src/types.rs#L23-L70 and in result I'm down to one case where I actually have something that will produce `???` and it's the scenario where the default variant is missing - https://github.com/projectfluent/fluent-rs/blob/master/fluent-bundle/src/types.rs#L149-L157 and https://github.com/projectfluent/fluent-rs/blob/master/fluent-bundle/src/resolve.rs#L206

I hope to move away from that as well and remove `FluentValue::None` completely.

I think it's a cleaner design with easier maintenance and ability to further improve UX of handling error scenarios.

Reply:
It also improved the experience when the error is `key0 = Foo { key1.attr }` to avoid formatting it into `Foo { key1 }` in the error scenario - https://github.com/projectfluent/fluent-rs/blob/master/fluent-bundle/tests/resolve_attribute_expression.rs#L44-L58

Reply:
> I hope to move away from that as well and remove FluentValue::None completely.

But you'll still have `FluentValue::Error`, right? There needs to be something for storing a failed resolution. In `fluent.js` it's `FluentNone` because error reporting is not very advanced right now. In this PR, I only want to improve the output of the fallback. Error reporting is a much larger task in scope and I'd prefer to first design the spec for it.

Reply:
> It also improved the experience when the error is key0 = Foo { key1.attr } to avoid formatting it into Foo { key1 } in the error scenario - https://github.com/projectfluent/fluent-rs/blob/master/fluent-bundle/tests/resolve_attribute_expression.rs#L44-L58

I have a branch which improves this, too :) I'll open a PR later today.

Reply:
See https://github.com/projectfluent/fluent.js/pull/370.